src/include/wiredtiger.in

```diff
@@ -4773,66 +4748,49 @@ struct __wt_location_handle {
-
-	/*!
-	 * Free memory allocated by WT_STORAGE_SOURCE::location_list.
-	 *
-	 * @errors
-	 *
-	 * @param storage_source the WT_STORAGE_SOURCE
-	 * @param session the current WiredTiger session
-	 * @param object_list array returned by WT_STORAGE_SOURCE::location_list
-	 * @param count count returned by WT_STORAGE_SOURCE::location_list
-	 */
-	int (*ss_location_list_free)(WT_STORAGE_SOURCE *storage_source,
-	    WT_SESSION *session, char **object_list, uint32_t count);
-
-	/*!
-	 * Return if the named object exists in the location.
-	 *
+	 * Create a customized file system to access the storage source
+         * objects.
+         *
+         * The file system returned behaves as if objects in the specified buckets
+         * are files in the file system.  In particular, the fs_open_file method
+         * requires its flags argument to include either WT_FS_OPEN_CREATE or
+         * WT_FS_OPEN_READONLY.  Objects being created are not deemed to "exist" and
+         * be visible to other APIs WT_FILE_SYSTEM::fs_exist until the new handle has been closed.
+	 * Objects once created are immutable. That is, only objects that do not already
+	 * exist can be opened with the create flag, and objects that already exist can
+	 * only be opened with the readonly flag.  Only objects that exist can be
+         * transferred to the underlying shared object storage.  This can happen at any
+         * time after an object is created, and can be forced to happen using
+         * WT_STORAGE_SOURCE::ss_flush.
+	 *
+	 * Additionally file handles returned by the file system behave as file handles to a
+         * local file.  For example, WT_FILE_HANDLE::fh_sync synchronizes writes to the
+         * local file, and does not imply any transferring of data to the shared object store.
+         *
+         * The directory argument to the WT_FILE_SYSTEM::fs_directory_list method is normally
+         * the empty string as the cloud equivalent (bucket) has already been given when
+         * customizing the file system.  If specified, the directory path is interpreted
+         * as another prefix, which is removed from the results.
+         *
+         * Names used by the file system methods are generally flat.  In particular, "..", ".",
+         * and "/" have no particular meaning, unlike POSIX file systems.
+         *
 	 * @errors
 	 *
 	 * @param storage_source the WT_STORAGE_SOURCE
 	 * @param session the current WiredTiger session
-	 * @param location_handle the location to search
-	 * @param name the name of the object
-	 * @param[out] existp If the named storage source object exists
+	 * @param bucket_name the name of the bucket.
+         * @param prefix a prefix for each file.  If used, the prefix will be added to the
+         *    name of each object created or otherwise accessed in the bucket.  Also, only
+         *    objects with this prefix will be visible, and the prefix will be removed when
+         *    listed.
```

Comment:
Above you say that `/` has no particular meaning, but in the tech design we're planning to append a `/` between bucket/prefix/name. Do you want to talk about mention that here?

Reply:
I think I was wrong to say that the name space is flat with "..", ".", "/" having no particular meaning.  That's an ideal, and it may be (kind of) true for AWS, but it won't be quite true for the local_storage implementation.  For example, if you created a prefix with the name of "../../../../../bin" that may be problematic!  For that section, I'll mention that there may be some reasonable restrictions.   I'll mention the use of `/` here as well.